controller/service.go

```diff
@@ -220,6 +221,16 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 		if serviceIPFamily != desiredLbIPFamily {
 			return nil, fmt.Errorf("requested loadBalancer IP(s) %q does not match the ipFamily of the service", desiredLbIPs)
 		}
+
+		nodes := []v1.Node{}
+		for _, node := range c.nodes {
+			nodes = append(nodes, *node)
+		}
+
+		nodeIPs := k8snodes.NodeIPsForFamily(nodes, serviceIPFamily)
+		if err = checkSvcIPsOverlapWithNodeIPs(desiredLbIPs, nodeIPs); err != nil {
```

Comment:
I would argue that this will never happen, because even if the service asks for a node ip, that ip will never be available in a pool. Let's remove the controller bits and focus only on the configuration part.

Reply:
ok. will remove.

Reply:
This comment is resolved. Checking for service LB ip against node IPs is removed.